### DISCUSSION TITLE
blog/2023/relay-tunnel/

### DISCUSSION THREAD

**Discussion by giscus** [go-gost/gost#575] [General]
# blog/2023/relay-tunnel/

https://gost.run/blog/2023/relay-tunnel/

**Comment by chenbuer** [comment]
成功了！

不过这种情况也只能穿透http/https服务。请教下，要是想像cloudflare tunnel那样穿透tcp服务，要怎么处理？cloudflare tunnel是在user端口再跑一个[warp或者cloudflared](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/use-cases/)。

求教，用gost怎么在user端增加一个entrypoint。万分感谢。

**Reply by chenbuer** [reply]
保留公共访问点（该篇博文中tunnel.my.domain指向的VPS）

**Comment by dream-lmc** [comment]
基于源码开发，我使用 Socks5 来代理udp流量时，发现每次返回 UDP的中继器都会在本地新创建一个随机的端口？请问这个优化一下吗，支持用户自定义Socks5 返回 UDP 固定域名和端口，同时增加共享一个UDP端口。

主要是为里解决 服务部署在 云服务器上时，NLB 负载均衡的问题。

不知道在哪里联系你。就发在这里了 @ginuerzh
